docs/blog/posts/pydantic.md

```diff
@@ -0,0 +1,303 @@
+
+    ```python
+    Person(name="Sam", age="13.4")
+    >>> ValidationError: 1 validation error for Person
+    >>> age
+    >>> Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='13.4', input_type=str]
+    >>>        For further information visit https://errors.pydantic.dev/2.5/v/int_parsing
+    ```
+
+    This behavior is great when we may not have trusted inputs, but is even more critical when inputs are coming from a language model!
+
+    To learn more about validation, check out the section [validation is a misnomer](https://docs.pydantic.dev/latest/concepts/models/#tldr)
+
+By Just prompting the model with the following prompt, we can generate a JSON schema for a `PythonPackage` dataclass.
+
+```python
+from pydantic import BaseModel
+from openai import OpenAI
+
+client = OpenAI()
+
+
+class Package(BaseModel):
+    name: str
+    author: str
+
+resp = client.chat.completions.create(
+    model="gpt-3.5-turbo",
+    messages=[
+        {
+            "role": "user",
+            "content": "Return the `name`, and `author` of pydantic, in a json object."
+        },
+    ]
+)
+
+Package.model_validate_json(resp.choices[0].message.content)
+```
+
+If all is well, we might get something that looks like `json.loads({"name": "pydantic", "author": "Samuel Colvin"})` which is correctly deserialized. But if something is wrong, we might get a whole bunch of text that contains prose or markdown code blocks that we'd have to reasonable
+
+**LLM responses with markdown code blocks**
+
+````python
+json.loads("""
+```json
+{
+  "name": "pydantic",
+  "author": "Samuel Colvin"
+}
+```
+""")
+>>> JSONDecodeError: Expecting value: line 1 column 1 (char 0
+````
+
+**LLM responses with prose**
+
+```python
+json.loads("""
+Ok heres the authors of pydantic: Samuel Colvin, and the name this library
+
+{
+  "name": "pydantic",
+  "author": "Samuel Colvin"
+}
+""")
+>>> JSONDecodeError: Expecting value: line 1 column 1 (char 0
+```
+
+The content may contain valid JSON, but it isn't considered valid JSON without understanding the language model's behavior. However, it could still provide useful information that we need to handle independently. Fortunately, `OpenAI` offers several options to address this situation.
+
+## Introducing Tools Calling
+
+While tool calling was originally designed to make calls to external APIs using JSON schema, its real value lies in allowing us to specify the desired output. Fortunately, `Pydantic` provides utilities for generating a JSON schema and supports nested structures, which would be difficult to describe in plain text.
+
+In this example, instead of describing the desired output in plain text, we can simply provide the JSON schema for the `Packages` class, which includes a list of `Package` objects.
```

Comment:
I changed the final period to a colon, because I think this should be followed immediately by the Pydantic-using example code. (I've added notes below about what to do with the things currently in between.)